common/src/gas_provider/auxiliary.rs

```diff
@@ -195,40 +208,49 @@ impl Error for GasTreeError {
     }
 }
 
-std::thread_local! {
-    // Definition of the `TotalIssuance` global storage, accessed by the tree.
-    pub(crate) static TOTAL_ISSUANCE: RefCell<Option<Balance>> = const { RefCell::new(None) };
+pub trait TotalIssuanceStorage<T: TotalIssuanceProvider> {
+    fn storage() -> &'static LocalKey<T>;
+}
+
+pub trait TotalIssuanceProvider {
+    fn data(&self) -> Ref<'_, Option<Balance>>;
+    fn data_mut(&self) -> RefMut<'_, Option<Balance>>;
 }
 
 /// Global `TotalIssuance` storage manager.
 #[derive(Debug, PartialEq, Eq)]
-pub struct TotalIssuanceWrap;
+pub struct TotalIssuanceWrap<TIStorage, TIProvider>(PhantomData<(TIStorage, TIProvider)>);
 
-impl ValueStorage for TotalIssuanceWrap {
+impl<TIStorage, TIProvider> ValueStorage for TotalIssuanceWrap<TIStorage, TIProvider>
```

Comment:
value storage could also be implemented generally for arbitrary wraps like RefCell<Option<T>> as well as mapping for RefCell<BTreeMap<T, V>>